app/scripts/metamask-controller.js

```diff
@@ -2787,6 +2788,40 @@ export default class MetamaskController extends EventEmitter {
       getPermittedAccountsByOrigin,
     );
 
+    this.controllerMessenger.subscribe(
```

Comment:
Would this big change be problematic before release as well?

Reply:
It shouldn't as there wouldn't be any way to add or modify the permittedChains permission, so this handler should effectively be one big noop.